packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -148,16 +148,19 @@ class AOTSnapshotter {
     // TODO(cbracken): replace IOSArch with TargetPlatform.ios_{armv7,arm64}.
     assert(platform != TargetPlatform.ios || iosArch != null);
 
-    final bool compileToSharedLibrary = preferSharedLibrary && androidSdk.ndkCompiler != null;
-    if (preferSharedLibrary && !compileToSharedLibrary) {
-      printStatus('Could not find NDK compiler. Not building in shared library mode.');
+    // buildSharedLibrary is Android-only.
+    assert(platform != TargetPlatform.ios || !buildSharedLibrary);
```

Comment:
what happens if you run with asserts disabled, build for iOS and we pass buildSharedLibrary by accident? 

can we simply make it a noop on iOS instead of an assertion?

Reply:
Done.